src/pydata_sphinx_theme/assets/styles/extensions/_notebooks.scss

```diff
@@ -9,84 +9,202 @@
+      // need find a different way to achieve this.
+      :not(.highlight, .widget-tab-contents, .jupyter-widgets):last-child {
+        margin-top: 0;
+        margin-bottom: 0;
+      }
+
+      div.highlight,
+      div.nohighlight,
+      .jupyter-widgets,
+      .jupyter-matplotlib-canvas-container {
+        overflow: visible; // for focus ring
+      }
+
+      // use theme color for stderr
       &.stderr {
-        background-color: var(--pst-color-danger);
+        background-color: transparent;
+
+        div[class*="highlight"] > pre {
+          background-color: var(--pst-color-danger-bg);
+        }
+      }
+
+      // Override our own style rule in _math.scss
+      div.math {
+        display: block;
+      }
+
+      span.math {
+        display: inline;
       }
     }
   }
-}
 
-// Add extra padding to the final item in an nbsphinx container
-div.nblast.container {
-  margin-bottom: 1rem;
-}
-
-// Override nbsphinx's colors for notebook cell prompts because they do not have
-// sufficient contrast. Colors chosen from accessible-pygments
-// a11y-high-contrast-{light,dark} themes.
+  // Override nbsphinx's styles with our own <pre> tag styles
+  div.nbinput.container div.input_area div[class*="highlight"] > pre,
+  div.nboutput.container div.output_area div[class*="highlight"] > pre {
+    @extend %pre-style;
 
-// Notebook cell input line number. Replace nbsphinx's low contrast blue with
-// higher contrast blues.
-.nbinput.container .prompt pre {
-  html[data-theme="light"] & {
-    // Copied from accessible-pygments [a11y-high-contrast-light](https://github.com/Quansight-Labs/accessible-pygments/tree/main/a11y_pygments/a11y_high_contrast_light)
-    color: #005b82;
+    margin: 0;
   }
 
-  html[data-theme="dark"] & {
-    // Copied from accessible-pygments [a11y-high-contrast-dark](https://github.com/Quansight-Labs/accessible-pygments/tree/main/a11y_pygments/a11y_high_contrast_dark)
-    color: #00e0e0;
+  // Last container styles
+  div.container.nblast {
+    margin-bottom: $top-and-bottom-container-space;
+
+    // Override nbsphinx's padding rule. Let the whitespace be controlled by the
+    // margin-bottom rule above. This is important because if we create
+    // whitespace with padding instead of margin, then we create a misalignment
+    // between the vertical left bar and the bottom of the notebook cell.
+    padding-bottom: 0;
   }
 }
 
-// Notebook cell output line number. Replace nbsphinx's low contrast red with
-// higher contrast red / orange.
-.nboutput.container .prompt pre {
-  html[data-theme="light"] & {
-    // Copied from accessible-pygments [a11y-high-contrast-light](https://github.com/Quansight-Labs/accessible-pygments/tree/main/a11y_pygments/a11y_high_contrast_light)
-    color: #a12236;
-  }
+// Nest under .bd-article-container so that selectors have higher specificity
+// and can therefore override the rules in the nbphinx stylesheet.
+.bd-article-container {
```

Comment:
Some of these rules used to be nested under `.bd-content`. But as far as I can tell, notebook content will always be rendered within `.bd-article-container`, so this is the better selector to use for nesting.